libecl/src/ecl_rst_file.c

```diff
@@ -73,6 +74,67 @@ ecl_rst_file_type * ecl_rst_file_open_read( const char * filename ) {
 }
 
 
+
+/*
+  This function will scan through the file and look for seqnum
+  headers, and position the file pointer in the right location to
+  start writing data for the report step given by @report_step. The
+  file is truncated, so that the filepointer will be at the (new) EOF
+  when returning.
+*/
+
+
+ecl_rst_file_type * ecl_rst_file_open_write_seek( const char * filename , int report_step) {
+  ecl_rst_file_type * rst_file = ecl_rst_file_alloc( filename  );
+  offset_type target_pos = 0;
+  bool seqnum_found = false;
+  rst_file->fortio = fortio_open_readwrite( filename , rst_file->fmt_file , ECL_ENDIAN_FLIP );
+  /*
+     If the file does not exist at all the fortio_open_readwrite()
+     will fail, we just try again - opening a new file in normal write
+     mode, and then immediately returning.
+  */
+  if (!rst_file->fortio) {
+    rst_file->fortio = fortio_open_writer( filename , rst_file->fmt_file , ECL_ENDIAN_FLIP );
+    return rst_file;
+  }
+
+  fortio_fseek( rst_file->fortio , 0 , SEEK_SET );
+  {
+    ecl_kw_type * work_kw = ecl_kw_alloc_new("WORK-KW" , 0 , ECL_INT_TYPE , NULL);
+
+    while (true) {
+      offset_type current_offset = fortio_ftell( rst_file->fortio );
+
+      if (fortio_read_at_eof(rst_file->fortio)) {
+        if (seqnum_found)
+          target_pos = current_offset;
+        break;
+      }
+
+      if (ecl_kw_fread_header( work_kw , rst_file->fortio)) {
```

Comment:
I honestly think this would be easier to follow if you do `if( !fread_header ) break`, since there are three more code paths inside this if that would lead to break, and with 2-space indentation it's quite hard to visually determine which else clause belongs to which condition.